ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -250,20 +299,21 @@ TestObjectExists(const Aws::S3Crt::ClientConfiguration &config)
     File << "Test payload";
     File.close();
 
-    if (ret = conn.ObjectExists(objectName, exists) != 0 || exists)
+    ret = conn.ObjectExists(objectName, exists);
+    if (ret != 0 || exists)
         return (ret);
 
-    if (ret = conn.PutObject(objectName, fileName) != 0)
+    if ((ret = conn.PutObject(objectName, fileName)) != 0)
         return (ret);
 
-    if (ret = conn.ObjectExists(objectName, exists) != 0 || !exists)
+    ret = conn.ObjectExists(objectName, exists);
+    if (ret != 0 || !exists)
         return (ret);
```

Comment:
Same as before. You want to return a failure even in case of "!exists", but this call will return a 0, which is success.